pkg/crc/image/image.go

```diff
@@ -0,0 +1,113 @@
+package image
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"runtime"
+	"strings"
+
+	crcpreset "github.com/code-ready/crc/pkg/crc/preset"
+	"github.com/containers/image/v5/copy"
+	"github.com/containers/image/v5/signature"
+	"github.com/containers/image/v5/transports/alltransports"
+	"github.com/containers/image/v5/types"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+var (
+	registryURI = "quay.io"
+	orgName     = "crcont"
+)
+
+type OCIImageHandler struct {
+	ImageName string
+	ImageTag  string
```

Comment:
Not clear to me why we hardcode `quay.io` and `crcont` but we let external code specify which `image`/`tag` should be downloaded, especially with `GetImageName` implemented in this file.

Reply:
Didn't get what you mean about external code as of now `GetImageName` only provide the image name for preset and then we define this struct and use the method to copy it.

Reply:
`preflight_checks_common.go` has code to fill ImageName/ImageTag:

```
imgHandler := image.OCIImageHandler{
		ImageName: image.GetImageName(preset),
		ImageTag:  version.GetCRCVersion(),
		Progress:  os.Stdout,
	}
```
but it calls into `image` to fill this.  Then `image.go` hardcodes `quay.io/crcont`.  Half of what is needed to fetch the image is set in `preflight_checks_common.go` and the other half in `image.go`. `preflight_checks_common.go` does not need to set this imo, ImageName/ImageTag could both be set internally in `image.go`